src/main/java/edu/kit/compiler/optimizations/NodeCollector.java

```diff
@@ -0,0 +1,20 @@
+package edu.kit.compiler.optimizations;
+
+import firm.nodes.Node;
+import firm.nodes.NodeVisitor;
+import lombok.Getter;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Getter
+public class NodeCollector extends NodeVisitor.Default {
```

Comment:
This class is currently not used. However, I would suggest to keep it and instead remove `WorklistFiller` and `DefaultWorklist`; the CSE doesn't seem to need a proper worklist anyway (it only iterates over the nodes in the list and doesn't reinsert any nodes).

Reply:
There are a number of visitors for filling data structures in `optimizations.Util`. We should probably use those instead.

Reply:
`DefaultWorklist` and `WorklistFiller` have been replaced with the `NodeCollector` instead

Reply:
Why create the new class `NodeCollector`, if we already have `Util.NodeListFiller`?